File: ejercicios hechos/Java/estructura secuencial/act7.java
/*Programa: act7.java
 *Propósito: Realiza un programa que reciba una cantidad de minutos y muestre por pantalla a cuantas horas y minutos corresponde.
 *Autor: Jose Manuel Serrano Palomo.
 *Fecha: 07/11/2019
 *
 *Análisis:
 * Debemos calcular a que equivale en horas y minutos, unos minutos introducidos por pantalla
 * 
 *Diseño:
 * Leemos min
 * Calculamos horas <-- min/60
 * Calculamos minsRes <-- min%60
 * Mostramos por pantalla
 * Variables: min son los minutos introducidos por el usuario. horas son el resultado en horas de los minutos y minRes es el resultado restante de los minutos
*/

import java.util.Scanner;

public class act7 {
  public static void main(String args[]) {
    Scanner s = new Scanner(System.in);
    System.out.println("Calculo de horas y minutos");
    System.out.println("--------------------------");
    int min;
    int horas;
    int minRes;

    System.out.println("Dime cuantos minutos son:");
    min = s.nextInt();

    horas = min/60;
    minRes = min%60;
    System.out.println("El resultado son "+horas+" horas y "+minRes+" minutos.");
  }


}

File: ejercicios terminados/Python/ejercicios repeticiones/act7.py
#Programa: act7.py
#Propósito: Una persona adquirió un producto para pagar en 20 meses. El primer mes pagó 10 €, el segundo 20 €, el tercero 40 € y así sucesivamente. Realizar un programa para determinar cuánto debe pagar mensualmente y el total de lo que pagará después de los 20 meses.
#Autor: Jose Manuel Serrano Palomo.
#Fecha: 29/10/2019
#
#Variables a usar:
# P es el pago inicial, pac es el total del pago tras acumularlo, mes es la cantidad de meses
#
#Algoritmo:
# Debemos ir acumulando los pagos en una serie aritmética de 20 veces multiplicando poe 2
# Desde el mes 1 al 20
#   Acumula el pago
#   Se dobla
# Mostramos el pago acumulado 

print("Calcular el pago acumulado de un comprador")
print("------------------------------------------\n")

# Leemos los datos

P = 10
pac = 0

#Hacemos la acumulación
for mes in range (1, 20):
    pac = pac + P
    P = P * 2

print(f"El resultado de los 20 meses al pagar es {pac}")
